docs/media-app.md

```diff
@@ -0,0 +1,99 @@
+For example, you won't be able to upload a PNG file with a `.jpg` extension, or vice-versa.
+This helps to prevent exceptions being thrown while thumbnailing images on the front-end of the site.
+
+#### Model fields
+
+* `title`: A name for the file.
+In the admin, this is prepopulated from the filename when first uploaded, if no title is provided by the user.
+* `labels`: A `ManyToManyField` to `Label` (see below).
+* `file`: A Django `FileField`, which is the file itself.
+* `attribution`, `copyright` and `alt_text`: Additional metadata fields.
+It is up to you how, or if, to render these on the front end of the site.
+
+In addition, the following fields are present on the model, but are not user-visible and are automatically populated on save:
+
+* `width` and `height`: The image dimensions of the file, if the file is an image.
+* `date_added`: The time the file was first uploaded. This is used for ordering in the admin (most recent first).
+
+#### Model methods & properties
+
+* `get_dimensions()`: If the file is an image, returns a tuple of (width, height), otherwise returns 0.
+This is only used internally; you probably want to access the `width` and `height` fields on the model instead, as they incur no overhead.
+* `icon`: A cached property that returns the path to an appropriate icon for the file type, e.g. `/static/media/img/x-office-spreadsheet.png`. This is used as a fallback in the media list if a file is not an image.
+* `is_image()`: Returns True if the file is an image (based on the file extension), False otherwise:
+
+### Label
+
+`cms.apps.media.models.Label` helps administrators organise media;
+think of them as tags, or notes to self.
+The are not intended to be shown to users on the front end of a website.
+
+Label has only one field: a `title`, which is also used as the ordering field.
+
+### Video
+
+`cms.apps.media.models.Video` is a collection of video files and related imagery.
+You can use it to easily create cross-browser compatible `<video>` tags on the frontend of your website.
+
+It has the following fields:
+
+* `title` is self-explanatory.
+* `external_video` is for embedding an off-site video (e.g. YouTube, Vimeo, etc). More on this below.
+* `high_resolution_mp4` is for a self-hosted MP4 (H.264). This is a `VideoFileRefField`, which is a reference to a `media.File`.
+* `low_resolution_mp4` for a lower-resolution version suitable for playing on mobile devices.
+* `image` is for displaying a cover image on the video (e.g. for the `poster` attribute on the `<video>` tag).
+
+For external videos, the `save` method on the model will work out how to embed the video.
+If the video has oEmbed information, some hidden fields will be populated that you can use on your front end to embed it:
+
+* `external_video_id` will be the provider's ID for the video.
+For YouTube, this will be a string of seemingly-random characters (e.g. `3LlAi8ygeMI`), or a number for Vimeo videos.
```

Comment:
Consider changing to "For YouTube, this will be a string of seemingly-random characters (e.g. `3LlAi8ygeMI`), for Vimeo this will be a number."

I couldn't find anywhere that says what the length of Vimeo ids are, which is annoying